src/bloodhound/transport.js

```diff
@@ -47,7 +47,10 @@ var Transport = (function() {
 
     _fingerprint: function fingerprint(o) {
       o = o || {};
-      return o.url + o.type + $.param(o.data || {});
+      //$.param(o.data || {}) crashes because it does not expect a single string value.
+      //if o.data is string no need to call param
+
+	  return o.url + o.type + (o.data &&_.isString(o.data) ? o.data : $.param(o.data || {}));
```

Comment:
Are you missing a space here? `(o.data &&_.isString(o.data)` seems like it should be `(o.data && _.isString(o.data)`.